client/src/pages/Signup.jsx

```diff
@@ -112,12 +124,24 @@ function useRegister() {
   const login = async (username, email, password) => {
     console.log(email, password);
     const res = await fetch(
-      `/auth/signup?username=${username}&email=${email}&password=${password}`
-    ).then(res => res.json());
+      `/auth/signup?username=${username}&email=${email}&password=${password}`,
+      {
+        method: "POST",
+        // headers: { "Content-Type": "application/json" },
+        // body: { "email": email, "username": username, "password": password }
+      }
```

Comment:
Since we're using POST to send the data, we should put all the data in the body rather than the query string. As a reminder, you can convert the necessary parameters to a JSON string using [`JSON.stringify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify).

Reply:
Thank you, Trey. I have modified this part in the new pull request for the Issue 2